### PR TITLE
Loyalty statistics module

### PR DESCRIPTION
**Pull request by ardier16**

### CODE REVIEW

File: src/vue/modules/loyalty-points-statistics/index.vue [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    flex: 0 1 calc(100% - 4rem);
+  }
+}
+
+.loyalty-points-statistics__volumes {
+  margin-top: 2rem;
+}
+
+.loyalty-points-statistics_viewer-title {
+  margin-bottom: 3.2rem;
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/index.vue]: 1-1.6rem looks better

File: src/vue/modules/loyalty-points-statistics/_api.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ let _api = null
 
 /**
  * @param {Wallet} wallet - wallet to sign the requests
- * @param {Wallet} config - wallet to sign the requests
+ * @param {Object} config
  * @param {Wallet} config.horizonURL - the url of the horizon server
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/_api.js]: `{String}`

File: src/vue/modules/loyalty-points-statistics/components/charts/pie-chart.vue [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  }
+
+  &__lines polyline {
+    opacity: 0.3;
+    stroke-width: 0.2rem;
+    stroke: $col-pie-chart-stroke;
+    fill: none;
+  }
+
+  &__label {
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/components/charts/pie-chart.vue]: Please, don't use this (&) with classes. This is a good feature, but it makes a search by class name a bit hard. In this case, you cannot search for your `pie-chart__lines` class by the application.

Comment (letehaha) [reply src/vue/modules/loyalty-points-statistics/components/charts/pie-chart.vue]: Also, the label's styles look like copy-pasted from other resource and does not fit into our design

File: src/vue/modules/loyalty-points-statistics/components/viewers/incoming-volume-viewer.vue [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    <line-chart
+      v-if="isLoaded"
+      id="incoming-volume-chart"
+      :data="incomingVolumes"
+      :currency="point"
+    />
+
+    <load-spinner
+      v-else-if="!isLoadFailed"
+      message-id="loyalty-points.statistics.loading-msg"
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/components/viewers/incoming-volume-viewer.vue]: Move these translations to the `incoming-volume-viewer` scope

File: src/vue/modules/loyalty-points-statistics/components/viewers/summary-volume-viewer.vue [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    <line-chart
+      v-if="isLoaded"
+      id="summary-volume-chart"
+      :data="summaryVolumes"
+      :currency="point"
+    />
+
+    <load-spinner
+      v-else-if="!isLoadFailed"
+      message-id="loyalty-points.statistics.loading-msg"
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/components/viewers/summary-volume-viewer.vue]: Read above about translation scope

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      :data="requestsStatistics"
+    />
+
+    <load-spinner
+      v-else-if="!isLoadFailed"
+      message-id="loyalty-points.statistics.loading-msg"
+    />
+
+    <p v-else>
+      {{ 'loyalty-points.statistics.loading-error-msg' | globalize }}
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue]: Read above about translation scope

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+import LoadSpinner from '@/vue/common/Loader'
+
+import withdrawalRequestsMock from '../../mocks/withdrawal-requests'
+
+import { WithdrawalRequest } from '../../wrappers/withdrawal-request'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+
+import { globalize } from '@/vue/filters/globalize'
+
+const COLORS = {
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue]: Colors of what?

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawals-viewer.vue [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  <div class="withdrawals-viewer">
+    <line-chart
+      v-if="isLoaded"
+      id="withdrawals-chart"
+      :data="withdrawals"
+    />
+
+    <load-spinner
+      v-else-if="!isLoadFailed"
+      message-id="loyalty-points.statistics.loading-msg"
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/components/viewers/withdrawals-viewer.vue]: Read above about translation scope

File: src/vue/modules/loyalty-points-statistics/index.vue [hunk 9]
```diff
@@ -0,0 +1,112 @@
+<template>
+  <div class="loyalty-points-statistics">
+    <div class="loyalty-points-statistics__withdrawals">
+      <div class="loyalty-points-statistics_viewer">
+        <h3 class="loyalty-points-statistics_viewer-title">
+          {{ 'loyalty-points.statistics.withdrawals-title' | globalize }}
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/index.vue]: Read above about translation scope

File: src/vue/modules/loyalty-points-statistics/index.vue [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+     * @property config.horizonURL - the url of horizon server (without version)
+     */
+    config: {
+      type: Object,
+      required: true,
+    },
+  },
+
+  data: _ => ({
+    isLoaded: false,
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/index.vue]: Useless variable

File: src/vue/modules/loyalty-points-statistics/index.vue [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+  flex-wrap: wrap;
+  margin: -2rem;
+}
+
+.loyalty-points-statistics_viewer {
+  margin: 2rem;
+  flex: 0 1 calc(50% - 4rem);
+  min-height: 100%;
+
+  @include respond-to-custom($x-medium) {
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/index.vue]: As I understood, this variable defined in the `variables.scss` that was imported to the `mixins.scss`? It's not crystal to understanding so we import `variables.scss` into the component styles too

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawals-viewer.vue [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+
+  methods: {
+    loadWithdrawals () {
+      try {
+        // TODO: load withdrawals from the API
+        const { data } = withdrawalsMock
+        this.withdrawals = data
+        this.isLoaded = true
+      } catch (e) {
+        this.isLoadFailed = true
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/components/viewers/withdrawals-viewer.vue]: Define this property into the `data` and set it in the `false` before starting your `try/catch`

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawals-viewer.vue [hunk 13]
```diff
@@ -0,0 +1,58 @@
+<template>
+  <div class="withdrawals-viewer">
+    <line-chart
+      v-if="isLoaded"
+      id="withdrawals-chart"
+      :data="withdrawals"
+    />
+
+    <load-spinner
+      v-else-if="!isLoadFailed"
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/components/viewers/withdrawals-viewer.vue]: `v-else-if="isLoadFailed"` and set here your `failed` message. Try not to use negation where it possible

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+  methods: {
+    loadWithdrawalRequests () {
+      try {
+        // TODO: load withdrawal requests from the API
+        const { data } = withdrawalRequestsMock
+        this.withdrawalRequests = data
+          .map(item => new WithdrawalRequest(item))
+        this.isLoaded = true
+      } catch (e) {
+        this.isLoadFailed = true
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue]: [Same](https://github.com/tokend/web-client/pull/142/files#diff-229a2ce66796ff4a50483b6fd6514cc6R52)

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue [hunk 15]
```diff
@@ -0,0 +1,92 @@
+<template>
+  <div class="withdrawal-requests-viewer">
+    <pie-chart
+      v-if="isLoaded"
+      id="requests-chart"
+      :data="requestsStatistics"
+    />
+
+    <load-spinner
+      v-else-if="!isLoadFailed"
```
Comment (letehaha) [inline src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue]: [Same](https://github.com/tokend/web-client/pull/142/files#diff-229a2ce66796ff4a50483b6fd6514cc6R10)

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue (hunk 14 above)
Comment (letehaha) [reply src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue]: Check your other chard renderers too

File: src/vue/modules/loyalty-points-statistics/index.vue (hunk 11 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/index.vue]: No, they are declared in the `mixins.scss` file

File: CHANGELOG.md [hunk 16]
```diff
@@ -32,6 +33,7 @@ for further information about branching and tagging conventions.
 - Trade chart displaying
 - Buttons position on Verify page
 - Drop down text overflow
+- Line chart displaying
```
[the remark this answers is not in the document]
Comment (ardier16) [reply CHANGELOG.md]: Renamed

File: src/vue/modules/loyalty-points-statistics/index.vue (hunk 1 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/index.vue]: Updated

File: src/vue/modules/loyalty-points-statistics/_api.js (hunk 2 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/_api.js]: Fixed

File: src/vue/modules/loyalty-points-statistics/components/charts/pie-chart.vue (hunk 3 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/components/charts/pie-chart.vue]: Updated

File: src/vue/modules/loyalty-points-statistics/components/viewers/incoming-volume-viewer.vue (hunk 4 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/components/viewers/incoming-volume-viewer.vue]: Moved

File: src/vue/modules/loyalty-points-statistics/components/viewers/summary-volume-viewer.vue (hunk 5 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/components/viewers/summary-volume-viewer.vue]: Moved

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue (hunk 6 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue]: Moved

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue (hunk 7 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue]: Renamed

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawals-viewer.vue (hunk 8 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/components/viewers/withdrawals-viewer.vue]: Moved

File: src/vue/modules/loyalty-points-statistics/index.vue (hunk 10 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/index.vue]: Removed

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawals-viewer.vue (hunk 12 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/components/viewers/withdrawals-viewer.vue]: Resolved

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawals-viewer.vue (hunk 13 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/components/viewers/withdrawals-viewer.vue]: Fixed

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue (hunk 15 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue]: Fixed

File: src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue (hunk 14 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/components/viewers/withdrawal-requests-viewer.vue]: Resolved

File: src/vue/modules/loyalty-points-statistics/index.vue (hunk 9 above)
Comment (ardier16) [reply src/vue/modules/loyalty-points-statistics/index.vue]: Moved

**Pull request merged**